README.md

```diff
@@ -568,10 +568,61 @@ Chart customization is available using the [Values](helm/values.yaml) file.
 
 ## Overall architecture
 
-Overall architecture with all main parts are displayed below:
+Overall architecture with all main parts is displayed below:
 
 ![Architecture diagram](docs/architecture_diagram.png)
 
+OpenShift LightSpeed service is based on FastAPI framework (Uvicorn) with Langchain for LLM interactions. The service is split into several parts described below.
+
+### FastAPI server
+
+Handles REST API requests from clients (mainly from UI console, but can be any REST API-compatible tool), handles requests queue, and also exports Prometheus metrics. Uvicorn framework is used as FastAPI implementation.
+
+### Authorization checker
+
+Manages authentication flow for REST API endpoints. Currently K8S/OCL-based authorization is used, but in future it will be implemented in a more modular way to allow registering other auth. checkers.
```

Comment:
but in **the** future it will be implemented...

Reply:
fixed, thank you